### PR TITLE
Connect to api feature

### PR DESCRIPTION
**Pull request by Mar12358**
### Added features:
- Bookstore API
- Usage of axios with createAsyncThunk
- Implement addBook and removeBook.

More info [here](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connect_to_api_v2_1.md).

### CODE REVIEW

File: src/components/BooksList.jsx [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
-
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect,  } from "react";
+import { getBooks } from "../redux/books/booksSlice";
 const BooksList = () => {
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getBooks());
+  });
```
Comment (victorokoroji) [inline src/components/BooksList.jsx]: - [x] Good job so far 👍🏽 . I can see that your getBooks action is making unlimited network requests. I think you should correctly update the dependency in your useEffect to fix this. e.g

```
  useEffect(() => {
    dispatch(getBooks());
  },[]);
```

**Review by victorokoroji — CHANGES_REQUESTED** [review]
Hi @Mar12358, 

You have done a great job so far! 💪 👏
You still need to work on a few issues before moving on to the next project, but you're almost there!

## Top highlight 🎯

- The Pull Request is following the Git Flow ✅ 
- CreateAsync thunk was properly used ✅ 
- Actions dispatch working✅ 
- APIs correctly called ✅ 


## Required Changes ♻️

- [ ] Kindly update the title in your readme. It says Math Magician instead of Bookstore

_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BooksList.jsx (hunk 1 above)
Comment (Mar12358) [reply src/components/BooksList.jsx]: I have a linter error with the dependencies array empty. should I add dispatch as dependency? or disable the eslint for that line? @vickymarz

**Review by divinecharlotte — APPROVED** [review]
# Hello @Mar12358     😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say  👍🏻 ✅ 
### To Highlight 💯


- Descriptive pr👍 
- Linters are passing👍 
 




## Congratulations! 🎉 



Cheers, and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**